### PR TITLE
Account blocking

### PR DESCRIPTION
**Pull request by gijskant**
- Lock account after 5 failed login attempts.
- Unlocking of accounts by administrator.
Fixes [PODIUM-20](https://jira.thehyve.nl/browse/PODIUM-20).

### CODE REVIEW

File: podium-uaa/src/test/resources/config/application.yml [hunk 1]
```diff
@@ -99,3 +99,7 @@ podium:
         contactEmail:
         license:
         licenseUrl:
+
+uaa:
+    security:
+        accountLockingPeriodSeconds: 3
```
Comment (berndvdveen) [inline podium-uaa/src/test/resources/config/application.yml]: Please add this property to other property files too

File: podium-gateway/src/test/javascript/spec/app/entities/organisation/organisation-detail.component.spec.ts [hunk 2]
```diff
@@ -58,14 +58,19 @@ describe('Component Tests', () => {
...
-            }).compileComponents();
+            }).compileComponents()
+            .then(() => {
+                fixture = TestBed.createComponent(OrganisationDetailComponent);
+                comp = fixture.componentInstance;
+                service = fixture.debugElement.injector.get(OrganisationService);
+            });
         }));
-
+/*
```
Comment (berndvdveen) [inline podium-gateway/src/test/javascript/spec/app/entities/organisation/organisation-detail.component.spec.ts]: Please remove

File: podium-uaa/src/main/java/org/bbmri/podium/validation/ValidPassword.java [hunk 3]
```diff
@@ -0,0 +1,17 @@
+package org.bbmri.podium.validation;
+
+import javax.validation.Constraint;
+import javax.validation.Payload;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+@Target({ElementType.METHOD, ElementType.FIELD})
```
Comment (gijskant) [inline podium-uaa/src/main/java/org/bbmri/podium/validation/ValidPassword.java]: For setters.

File: podium-uaa/src/test/resources/config/application.yml (hunk 1 above)
Comment (gijskant) [inline podium-uaa/src/test/resources/config/application.yml]: Well, isn't the default is okay?

File: podium-uaa/src/main/resources/i18n/messages.properties [hunk 4]
```diff
@@ -46,3 +46,5 @@ email.signature.servicelink=Selfservice: http://trait.topdesk.net
 email.signature.telephone=Telefoon: +31 (0)881167500
 
 email.autogenerated=This is an automatically generated message. If you have any questions, please send an email to servicedesk@ctmm-trait.nl
+
+podium.ValidPassword.message=The password is invalid.
```
Comment (gijskant) [inline podium-uaa/src/main/resources/i18n/messages.properties]: The format is invalid?

Comment (berndvdveen) [inline podium-uaa/src/main/resources/i18n/messages.properties]: That is what the validator does correct?

File: podium-uaa/src/test/resources/config/application.yml (hunk 1 above)
Comment (berndvdveen) [inline podium-uaa/src/test/resources/config/application.yml]: It is, however if it isnt mentioned in the properties file then its not clear that that value is configurable

File: podium-uaa/src/main/resources/i18n/messages.properties (hunk 4 above)
Comment (gijskant) [inline podium-uaa/src/main/resources/i18n/messages.properties]: "Password is correct" / "Password is incorrect" seems to be the right feedback for login, "Password is invalid" seems to be right feedback for an error when updating a password.

File: podium-uaa/src/main/resources/config/application.yml [hunk 5]
```diff
@@ -85,3 +85,9 @@ podium:
     ribbon:
         # comma separated list of profiles to show a ribbon on the GUI
         displayOnActiveProfiles: dev
+
+uaa:
+    security:
+        maxFailedLoginAttempts:
```
Comment (gijskant) [inline podium-uaa/src/main/resources/config/application.yml]: @berndvdveen This does not work. Any idea how an empty entry could be added?
This results in `2017-02-23 16:12:09.981 ERROR 5861 --- [           main] o.s.b.b.PropertiesConfigurationFactory   : Field error in object 'uaa' on field 'security.maxFailedLoginAttempts': rejected value []; codes [typeMismatch.uaa.security.maxFailedLoginAttempts,typeMismatch.security.maxFailedLoginAttempts,typeMismatch.maxFailedLoginAttempts,typeMismatch.long,typeMismatch]; arguments [org.springframework.context.support.DefaultMessageSourceResolvable: codes [uaa.security.maxFailedLoginAttempts,security.maxFailedLoginAttempts]; arguments []; default message [security.maxFailedLoginAttempts]]; default message [Failed to convert property value of type 'java.lang.String' to required type 'long' for property 'security.maxFailedLoginAttempts'; nested exception is org.springframework.core.convert.ConverterNotFoundException: No converter found capable of converting from type [java.lang.String] to type [long]]`

Comment (berndvdveen) [inline podium-uaa/src/main/resources/config/application.yml]: @gijskant http://docs.spring.io/spring-boot/docs/current/api/org/springframework/boot/context/properties/ConfigurationProperties.html

One way would be to use 'ignoreInvalidFields' when the values argumented (String -> long) cannot be coerced and it should default to the set value in the UaaProperties.java.

This is the reason that the default properties for the Async configuration (which are also defaulted with values in PodiumProperties.java) are also explicitly added as values in the application.yml 

Another way would be to also define the default values in the application.yml for the uaa properties.

Not sure if this is a SpringConfiguration bug or not.

**Pull request merged**